week-3/stretch/rotate-char.js

```diff
@@ -16,21 +16,66 @@
 // Given a lowercase letter character (char) and a positive integer shift,
 // When the function is called with these inputs,
 // Then it should rotate the lowercase letter by shift positions within the lowercase alphabet, wrapping around if necessary, and return the rotated lowercase letter as a string.
+
+function rotateCharacter(char, shift) {
+  const alphabet = "abcdefghijklmnopqrstuvwxyz";
+  const rotatedIndex = (alphabet.indexOf(char) + shift) % 26;
+
+  return alphabet[rotatedIndex];
+}
+
 console.log(rotateCharacter("a", 3)); // Output: "d"
 console.log(rotateCharacter("f", 1)); // Output: "g"
+console.log(rotateCharacter("x", 5)); // Output: "c"
+console.log(rotateCharacter("z", 10)); // Output: "j"
+
+/* for me : When indexOf(char) is 2 (representing 'c' in the alphabet) and shift is 5:
+
+indexOf(char) + shift equals 2 + 5 = 7.
+The modulo operation 7 % 26 results in 7.
+Counting seven positions from the starting point (index 0), it refers to the character at index 7 within the alphabet.*/
 
 // Scenario: Rotate Uppercase Letters:
 // Given an uppercase letter character (char) and a positive integer shift,
 // When the function is called with these inputs,
 // Then it should rotate the uppercase letter by shift positions within the uppercase alphabet, wrapping around if necessary, and return the rotated uppercase letter as a string.
-console.log(rotateCharacter("A", 3)); // Output: "D"
-console.log(rotateCharacter("F", 1)); // Output: "G"
+
+// The function takes a character (char) and a shift value (shift)
+
+// 1 Check if the character is in the lowercase alphabet
+// 2 Find the index of the character in the lowercase alphabet
+//  3 Calculate the new index after rotation
+// 4 Return the rotated character
+// 5 Check if the character is in the uppercase alphabet
+// 6 Calculate the new index after rotation
+// 7 Return the rotated character
+// 8 If the character is not a letter, return it unchanged
+
+function rotateCharacter(char, shift) {
+  const lowercaseAlphabet = "abcdefghijklmnopqrstuvwxyz";
+  const uppercaseAlphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+  if (lowercaseAlphabet.includes(char)) {
+    // if  lowercase letter
+    const rotatedIndex = (lowercaseAlphabet.indexOf(char) + shift) % 26; // read for me comment
+    return lowercaseAlphabet[rotatedIndex];
+  } else if (uppercaseAlphabet.includes(char)) {
+    const rotatedIndex = (uppercaseAlphabet.indexOf(char) + shift) % 26; // same for upperAlphabet
+    return uppercaseAlphabet[rotatedIndex];
+  } else {
+    return char; // (non-letter remains unchanged)
+  }
```

Comment:
This is good and definitely should work. Though it may be optimised in a few ways:

1) For now each time we call the function `rotateCharacter` we check/iterate the same string of letters 2 times:
- 1st time when we call `includes`
- 2nd time when we call `indexOf`
Can we avoid using `includes` at all in this task and just use only `indexOf` and only once?
What this method returns us when there is no value we search in string?

2) In the worst case (e.g. the argument is a number) we will check 2 strings of letters before returning original value back. But I think we can use one string of letters (e.g. `lowercaseAlphabet`) instead of two if we utilise the string methods `toLowerCase()` and `toUpperCase()`